pkgs/applications/misc/hello/default.nix

```diff
@@ -23,4 +23,6 @@ stdenv.mkDerivation rec {
     maintainers = [ maintainers.eelco ];
     platforms = platforms.all;
   };
-}
+
+  passthru.tests.run = callPackage ./test.nix { hello = this; };
```

Comment:
i don't really understand why this is an improvement.
wouldn't it have the same behavior when you remove `hello = this;`?
like in this example https://github.com/NixOS/nixpkgs/blob/d600f006643e074c2ef1d72e462e218b647a096c/pkgs/tools/misc/phoronix-test-suite/default.nix#L23

Reply:
That way you'll be testing the un-overridden version if you use `overrideAttrs`.

```
nix-instantiate --expr '((import ./. {}).phoronix-test-suite.overrideAttrs (o: { dontBuild = true; })).tests.simple-execution' '(import ./. {}).phoronix-test-suite.tests.simple-execution'
/nix/store/aj6lv43p8hlkhlbd6klljgami4cb48qv-phoronix-test-suite-tests.drv
/nix/store/aj6lv43p8hlkhlbd6klljgami4cb48qv-phoronix-test-suite-tests.drv
```

So, same test although the first one had `overrideAttrs` applied to it. Should have been different.

Reply:
i see. thanks